x-pack/plugin/otel-data/src/main/resources/component-templates/otel@mappings.yaml

```diff
@@ -0,0 +1,67 @@
+---
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for all OTel data streams
+  managed: true
+template:
+  mappings:
+    date_detection: false
+    dynamic: strict
+    properties:
+      "@timestamp":
+        type: date_nanos
```

Comment:
Because ES|QL doesn't support `date_nanos` at the moment (see https://github.com/elastic/elasticsearch/issues/109352), I think we need to revert to using `date` for now, until ES|QL reaches an acceptable level of support for `date_nanos`.

@mlunadia @strawgate @tylerperk, we could use your help to define what acceptable means and what level of parity we need in order for us to be ready to use the `date_nanos` field type. Do we need the experience with `date_nanos` to not be worse in any aspects compared to `date`? Are we willing to make compromises, for example users having to make manual casts in ES|QL? What are the essentials and what are the nice to haves?

Reply:
Would it make sense to default to `date` here but override it to `date_nanos` for specific datasets? I think most integrations do not use `date_nanos` today and it carries with it perf/storage costs?

Reply:
They don't but we're also discussing changing this as a default for integrations: https://github.com/elastic/logs-dev/issues/25

There's minimal overhead if `date_nanos` is used as a field type but the timestamps are only in ms precision. As OTel timestamps are defined in [nanosecond in OTLP](https://github.com/open-telemetry/opentelemetry-proto/blob/484241a016d7b81f20b6e19d00ffbc4a3b864a22/opentelemetry/proto/logs/v1/logs.proto#L140), I think we should also map them this way, without losing precision, by choosing the `date_nanos` field type by default.

Users can still override this and chose `date` instead.

However, in the short term, because of the limitations in ES|QL, I don't think we can use the `date_nanos` field type, yet.